wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,191 @@
+import django_filters
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db.models import Count, OuterRef, Q, Subquery
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.views.reports import ReportView
+from wagtail.coreutils import get_content_languages
+from wagtail.models import ContentType, Page, PageLogEntry, Site, get_page_models
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _get_site_choices():
+    choices = [(site.pk, str(site)) for site in Site.objects.all()]
+    return choices
+
+
+def _annotate_last_edit_info(queryset):
+    latest_edit_log = PageLogEntry.objects.filter(content_type=OuterRef("pk")).order_by(
+        "-timestamp", "-pk"
+    )
+
+    queryset = queryset.annotate(
+        count=Count("pages"),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+def _annotate_last_edit_info_by_locale(queryset, language_code):
+    latest_edit_log = PageLogEntry.objects.filter(
+        content_type=OuterRef("pk"), page__locale__language_code=language_code
+    ).order_by("-timestamp", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(pages__locale__language_code=language_code)),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
```

Comment:
Is there a way to come at this from the `Page` model side instead? That way, we might be able to drop some of the code further down.

Reply:
I wish we could, but the `related_name` from the `PageLogEntry` [prohibits the reverse lookup](https://github.com/wagtail/wagtail/blob/85a4a9ebf744c4f4881744a26f3e48262da2543e/wagtail/models/__init__.py#L4646-L4652):

```python
class PageLogEntry(BaseLogEntry):
    page = models.ForeignKey(
        "wagtailcore.Page",
        on_delete=models.DO_NOTHING,
        db_constraint=False,
        related_name="+",
    )
...
```

I'm open to suggestions, though.

Reply:
I see no reason why there shouldn't be a usable `related_name` value on that field, TBH. Maybe there's some kind of custom accessor on the page model side that the implementor didn't want it to be confused with?

Reply:
@jhonatan-lopes actually, maybe we don't really need the link table at all, and use `Page.owner` for the user column (and labelled accordingly) and `Page.latest_revision_created_at` as the datetime?